File: src/main/java/com/back/mozu/domain/reservation/service/WaitingRoomService.java
package com.back.mozu.domain.reservation.service;

import com.back.mozu.domain.reservation.dto.WaitingRoomResponseDto;
import com.back.mozu.domain.reservation.entity.Reservation;
import com.back.mozu.domain.reservation.entity.ReservationStatus;
import com.back.mozu.domain.reservation.repository.ReservationRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class WaitingRoomService {
    private final ReservationRepository reservationRepository;

    public WaitingRoomResponseDto getMyWaiting(UUID userId){
        Optional<Reservation> reservation = reservationRepository.findByUserIdAndStatus(
                userId, ReservationStatus.PENDING  // String → Enum으로 변경
        );
        return reservation.map(r -> WaitingRoomResponseDto.builder()
                .reservationId(r.getId())
                .status(r.getStatus().name())  // Enum → String으로 변환
                .queueNumber(null)
                .estimatedWaitMinutes(null)
                .build()
        ).orElse(null);
    }
}

File: src/main/java/com/back/mozu/domain/reservation/service/WaitingRoomService.kt
package com.back.mozu.domain.reservation.service

import com.back.mozu.domain.reservation.dto.WaitingRoomResponseDto
import com.back.mozu.domain.reservation.entity.ReservationStatus
import com.back.mozu.domain.reservation.repository.ReservationRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class WaitingRoomService(
    private val reservationRepository: ReservationRepository,
) {

    fun getMyWaiting(userId: UUID): WaitingRoomResponseDto? {
        val reservation = reservationRepository.findByUserIdAndStatus(
            userId, ReservationStatus.PENDING
        ) ?: return null

        return WaitingRoomResponseDto(
            reservationId = reservation.id,
            status = reservation.status?.name,
            queueNumber = null,
            estimatedWaitMinutes = null,
        )
    }
}
